coordinator.go

```diff
@@ -44,6 +44,9 @@ type Coordinator interface {
 	//
 	// Do not release tasks. The consumer will handle task releasing.
 	Close()
+
+	// String should return a name for the coordinator to use in logging.
+	String() string
```

Comment:
Ugh, this feels incredibly hacky, but it does allow contextual logging. Let me know if you have any better ideas or just think I should drop it until a better solution can be figured out... this is feels weird.